firestore/integration_test_internal/src/firestore_test.cc

```diff
@@ -87,6 +87,20 @@ TEST_F(FirestoreIntegrationTest, GetInstance) {
   delete auth;
 }
 
+TEST_F(FirestoreIntegrationTest, GetInstanceWithNamedDatabase) {
+  App* app = this->app();
+  EXPECT_NE(nullptr, app);
+
+  InitResult result;
+  Firestore* instance = Firestore::GetInstance(app, "foo", &result);
+  EXPECT_EQ(kInitResultSuccess, result);
+  EXPECT_NE(nullptr, instance);
+  EXPECT_EQ(app, instance->app());
+  // TODO(Mila): check the database name is "foo".
```

Comment:
Need more code changes to be able to read the database id out off the firestore instance

Reply:
Are you following up on this?